src/GitVersionExe/Program.cs

```diff
@@ -129,17 +130,27 @@ static void ConfigureLogging(Arguments arguments)
             {
                 try
                 {
-                    Directory.CreateDirectory(Path.GetDirectoryName(arguments.LogFilePath));
-                    if (File.Exists(arguments.LogFilePath))
+                    var logFileFullPath = Path.GetFullPath(arguments.LogFilePath);
+                    var logFile = new FileInfo(logFileFullPath);
+
+                    // NOTE: logFile.Directory will be null if the path is i.e. C:\logfile.log. @asbjornu
+                    if (logFile.Directory != null)
+                    {
+                        logFile.Directory.Create();
+                    }
+
+                    if (!logFile.Exists)
```

Comment:
I wouldn't worry about the squash. I don't mind seeing the process people take to get to the end solution. It is just forward merges which I like to see rebased out.

Lets just remove the if statement around this. Then it will either create it or empty the log file?

Or invert the if to put back to the old logic